### DISCUSSION TITLE
Feature: Hide image in editor

### DISCUSSION THREAD

**Discussion by miag676** [facebook/lexical#3163] [Q&A] [answered]
Is it possible to display the uploaded image as a link in the editor, instead of displaying the image in the editor. 
(Similar to StackOverflow and Github, where upon uploading the image, only the link is shown in the editor).

In this example I would like to display the link of the photo instead of the photo.

![Zajeta slika](https://user-images.githubusercontent.com/72701282/194805826-c448712c-5803-4000-bc57-5122dab49d06.PNG)

**Comment by acywatson** [comment, marked as answer]
Yea, you can render anything you want - you can just insert a LinkNode instead of an ImageNode. 

Here's some pseudocode:

```
editor.registerCommand('INSERT_IMAGE_COMMAND',(payload) => {
  const link = $createLinkNode(payload.href);
  const text = $createTextNode('My Image');
  link.append(text);
  selection.insertNodes([link]);
})
```

**Reply by miag676** [reply]
Thank you very much for the answer. It inserts the image as link, however when I write anything else in the editor, I get the error: `Expected node {number} to have a top parent element`. Could you please help me figure out what I did wrong?

This is my command:
```
editor.registerCommand(
        INSERT_IMAGE_COMMAND,
        (payload) => {
        const selection = $getSelection();
        if ($isRangeSelection(selection)) {
        if ($isRootNode(selection.anchor.getNode())) {
        selection.insertParagraph();
        }
        const link = $createLinkNode(payload.src);
        const text = $createTextNode('[Image]');
        link.append(text);
        selection.insertNodes([link]);
        }
        return true;
        },
        COMMAND_PRIORITY_EDITOR,
        )
```
